events/importer/espoo.py

```diff
@@ -425,6 +425,36 @@ def _add_id_to_dict(container: dict, obj, key):
             container[row["id"]] = row
 
 
+def purge_orphans(events_data: list[dict], skip_log=False) -> list[dict]:
+    """
+    This function can (and should) be called recursively to get rid of
+    broken super event references. Each iteration produces a map of known
+    events and then checks if there are any events that reference missing events.
+
+    Multiple iterations are required since each call gets rid of one "generation"
+    """
+    ids = [event_data["id"] for event_data in events_data]
+    keep = []
+    drop = set()
+    for event_data in events_data:
+        if (super_event := event_data["super_event"]) and (
+            super_event_id := parse_jsonld_id(super_event)
+        ) not in ids:
+            drop.add(super_event_id)
+            continue
+        keep.append(event_data)
+
+    if drop:
+        if not skip_log:
+            logger.error(
+                f"References to super events missing: {', '.join(drop)}",
```

Comment:
Perhaps we could rephrase something like:
```suggestion
                f"Events referenced as super events are missing from the data: {', '.join(drop)}",
```
or something like that.
Since we have the reference but not the actual data to which the data is referring to if I'm assuming things correct 🧐